### PR TITLE
Added option input-page-ranges in pdftopdf filter

### PR DESCRIPTION
**Pull request by vermamohit13**
Added orig-page-range option in pdftopdf filter which will provide more flexibility to the user for printing pages in a certain range. Unlike the existing page-range option, which applies the page-range parameter to the document after it has been processed, the orig-page-ranges option applies page-range before the document has been processed.
This option will also help in the solving the need of user as mentioned in issue #365.

### CODE REVIEW

File: cupsfilters/pdftopdf/pdftopdf_processor.cc [hunk 1]
```diff
@@ -30,6 +30,17 @@ bool ProcessingParameters::withPage(int outno) const // {{{
...
+{
+  if (pageno%2 == 0) { // 1-based
+    if (!evenPages) {
+      return false;
+    }
+  } else if (!oddPages) {
+    return false;
+  }
+  return origPageRange.contains(pageno);
+}
```
Comment (tillkamppeter) [inline cupsfilters/pdftopdf/pdftopdf_processor.cc:44]: Note that page-set=even and page-set=odd applies to the output pages as the original page-ranges attribute. The page-set attribute is for manual duplex (print odd pages, turn over the stack, put back into input tray, and print even pages). orig-page-ranges should simply select the pages from the input stream and the resulting list of pages passed on for all the other functionality of pdftopdf() to be applied.
If you have a 20-page input document and apply "orig-page-ranges=1-5,7-8,13-15 number-up=2 page-set=odd page-ranges=1-4" you get 2 sheets, the first contains the original pages 1 and 2, the second contains the original pages 5 and 7. Note that page-set and page-ranges are applied simultaneously. The page must be selected by both page-set and page-ranges to get finally selected. In our example we select 10 pages from the original 20 pages, then we put two on each sheet, ending up with 5 sheets, containing 1+2, 3+4, 5+7, 8+13, 14+15, after that we want to have the sheets 1, 2, 3,, and 4 but only the odd ones, so it remains sheet 1 and sheet 3, containing 1+2 and 5+7.

**Comment by tillkamppeter** [comment]
Another point: Let us rename from orig-page-ranges to input-page-ranges, as we treat the input pages which go into pdftopdf().

File: cupsfilters/pdftopdf/pdftopdf_processor.cc (hunk 1 above)
Comment (vermamohit13) [reply cupsfilters/pdftopdf/pdftopdf_processor.cc:37]: Thanks for bringing this to my attention.I made the necessary adjustments, and the option now functions as you indicated.

**Comment by vermamohit13** [comment]
> Another point: Let us rename from orig-page-ranges to input-page-ranges, as we treat the input pages which go into pdftopdf().

I have changed the name of the option to input-page-ranges.

**Comment by tillkamppeter** [comment]
@vermamohit13 thanks a lot, it works perfectly for me. An important missing feature is added now.

**Pull request merged**